### PR TITLE
[Issuse#4466] sync data support apollo

### PR DESCRIPTION
**Pull request by misaya295**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-bootstrap/src/main/resources/application.yml [hunk 1]
```diff
@@ -116,47 +116,47 @@ shenyu:
...
-#      handshakeTimeout:
-#      closeNotifyFlushTimeout:
-#      closeNotifyReadTimeout:
-#      defaultConfigurationType:
-#    threadPool:
-#      prefix: shenyu
-#      selectCount: 1
-#      workerCount: 8
-#      daemon: true
+  #  httpclient:
```
Comment (yu199195) [inline shenyu-bootstrap/src/main/resources/application.yml:119]: not modify this

File: shenyu-bootstrap/src/main/resources/application.yml [hunk 2]
```diff
@@ -167,19 +167,19 @@ shenyu:
     allowedHeaders:
     allowedMethods: "*"
     allowedAnyOrigin: true # the same of Access-Control-Allow-Origin: "*"
-#    allowedOrigin:
+      #    allowedOrigin:
```
Comment (yu199195) [inline shenyu-bootstrap/src/main/resources/application.yml:170]: not modify this

File: shenyu-examples/shenyu-examples-http/src/main/resources/application.yml [hunk 3]
```diff
@@ -20,12 +20,37 @@ server:
...
+#      connectionTimeout: 2000
+#      checked: true
+#      zombieCheckThreads: 10
+#      zombieCheckTimes: 5
+#      scheduledTime: 10
+#      nacosNameSpace: cab5cbe-d9d4-4a65-81f0-89d7511fdc7c
   register:
-    registerType: http #zookeeper #etcd #nacos #consul
-    serverLists: http://localhost:9095 #localhost:2181 #http://localhost:2379 #localhost:8848
+    registerType: apollo #http #zookeeper #etcd #nacos #consul
```
Comment (yu199195) [inline shenyu-examples/shenyu-examples-http/src/main/resources/application.yml:35]: not modify this

File: shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/proxy/ApacheDubboProxyService.java [hunk 4]
```diff
@@ -91,11 +91,10 @@ public Mono<Object> genericInvoker(final String body, final MetaData metaData, f
         })).onErrorMap(exception -> exception instanceof GenericException ? new ShenyuException(((GenericException) exception).getExceptionMessage()) : new ShenyuException(exception));
     }
     
-    @SuppressWarnings("unchecked")
     private CompletableFuture<Object> invokeAsync(final GenericService genericService, final String method, final String[] parameterTypes, final Object[] args) throws GenericException {
         //Compatible with asynchronous calls of lower Dubbo versions
         genericService.$invoke(method, parameterTypes, args);
-        Object resultFromFuture = RpcContext.getContext().getFuture();
-        return resultFromFuture instanceof CompletableFuture ? (CompletableFuture<Object>) resultFromFuture : CompletableFuture.completedFuture(resultFromFuture);
+        CompletableFuture<Object> future = RpcContext.getServiceContext().getCompletableFuture();
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/proxy/ApacheDubboProxyService.java:97]: All  apache  dubbo version have this method？

File: shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/ApacheDubboPlugin.java [hunk 5]
```diff
@@ -63,15 +63,15 @@ protected Mono<Void> doDubboInvoker(final ServerWebExchange exchange,
                                         final RuleData rule,
                                         final MetaData metaData,
                                         final String param) {
-        RpcContext.getContext().setAttachment(Constants.DUBBO_SELECTOR_ID, selector.getId());
-        RpcContext.getContext().setAttachment(Constants.DUBBO_RULE_ID, rule.getId());
-        RpcContext.getContext().setAttachment(Constants.DUBBO_REMOTE_ADDRESS, Objects.requireNonNull(exchange.getRequest().getRemoteAddress()).getAddress().getHostAddress());
+        RpcContext.getClientAttachment().setAttachment(Constants.DUBBO_SELECTOR_ID, selector.getId());
```
[the remark this answers is not in the document]
Comment (misaya295) [reply shenyu-plugin/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/ApacheDubboPlugin.java]: It should be submitted incorrectly. this pr does not involve changes to dubbo. I will fix it

**Review by li-keguo — APPROVED** [review]

**Review by yu199195 — APPROVED** [review]

**Pull request merged**